src/app/src/components/DrawMapUIComponents/EditToolbar.js

```diff
@@ -110,17 +134,23 @@ export default function EditToolbar() {
                         />
                         <EditToolbarButton
                             icon={TrashIcon}
-                            onClick={openConfirmDeleteDialog}
+                            onClick={confirmDeleteDialogController.open}
                             disabled={!polygon}
                         />
                     </ButtonGroup>
                 </Flex>
-                <DeletePolygonConfirmModal
-                    isOpen={showConfirmDeleteDialog}
-                    onConfirm={deletePolygon}
-                    onClose={closeConfirmDeleteDialog}
-                />
             </Box>
+            <DeletePolygonConfirmModal
+                isOpen={confirmDeleteDialogController.isOpen}
+                onConfirm={deletePolygon}
+                onClose={confirmDeleteDialogController.close}
+            />
+            <EditPolygonModal
+                isOpen={editDialogController.isOpen}
+                onSubmit={editPolygon}
+                onClose={editDialogController.close}
```

Comment:
This may read better as:

```suggestion
                isOpen={editDialog.isOpen}
                onSubmit={editPolygon}
                onClose={editDialog.close}
```

Reply:
My concern with this change is that `editDialog` doesn't actually represent a dialog. It's basically a container for variables and functions that control the state of the dialog. The dialog itself is represented by the JSX component to which the controller properties are passed.